### PR TITLE
WHFHRI-671 : Migrate Travis builds to GitHub Actions

### PR DESCRIPTION
**Pull request by alisakotliarova**
## PR Requirements:
 - [ ] The code must be reviewed by at least one Maintainer
 - [ ] JIRA Ticket: https://jira.wh-sdlc.watson-health.ibm.com/browse/WHFHRI-671
 - [ ] 90 % test code coverage
 - [ ] README updates

### CODE REVIEW

**Review by David-N-Perkins — APPROVED** [review]

**Review by aramcodz — CHANGES_REQUESTED** [review]
I believe that the build.gradle is missing the `ossIndexAudit{}` function block. 

See line 82 in the hir-flink-validation-fhir/build.gradle on the WHFHRI-672 branch (https://github.com/Alvearie/hri-flink-validation-fhir/blob/94c53c71e004bbd0271af2bd2220878dc67d873e/build.gradle#L82).

Also, if you have not already, you should add the `SONATYPE_OSS_PASSWORD` secret to the flink-pipeline-core repo. You can find that password value in 1Password ("Sonatype OSS Index Hribuild account)

**Review by aramcodz — APPROVED** [review]
Looks Good!

File: src/main/scala/org/alvearie/hri/api/MgmtClient.scala [hunk 1]
```diff
@@ -18,14 +18,17 @@ import com.fasterxml.jackson.databind.json.JsonMapper
...
 import java.util.Base64
 import java.util.ArrayList
 import org.slf4j.LoggerFactory
 
+import java.io.{File, FileNotFoundException}
 import scala.util.{Failure, Success, Try}
 
 class MgmtClient(val baseUri: String, val clientId: String, val clientSecret: String, val audience: String, val oauthServiceBaseUrl: String,
-                 val httpClient: CloseableHttpClient = HttpClients.createDefault()) extends Serializable with BatchLookup {
+                 val httpClient: CloseableHttpClient = MgmtClient.createHttpClient()) extends Serializable with BatchLookup {
```
Comment (aramcodz) [inline src/main/scala/org/alvearie/hri/api/MgmtClient.scala:31]: This change at line 31 is Not correct.

See the Private Repo code for `support-2.x` branch here --> https://github.ibm.com/wffh-hri/flink-pipeline-core/blob/support-2.x/src/main/scala/com/ibm/watson/health/foundation/hri/api/MgmtClient.scala

File: src/main/scala/org/alvearie/hri/api/MgmtClient.scala [hunk 2]
```diff
@@ -159,4 +162,44 @@ object MgmtClient {
...
+      throw new IllegalArgumentException(msg)
+    }
+    log.info("Creating Http client with trust store {}", trustStorePath)
+
+    val trustStoreFile = new File(trustStorePath);
+    if (!trustStoreFile.exists() || !trustStoreFile.isFile) {
+      val msg = "Not found or not a file: " + trustStoreFile.getPath
+      log.error(msg);
+      throw new FileNotFoundException(msg);
+    }
```
Comment (aramcodz) [inline src/main/scala/org/alvearie/hri/api/MgmtClient.scala:195]: This new method should not be included in the `support-2.x` branch of the code.

You are going to want to compare this alvearie versin of the file with the private Repo version --> https://github.ibm.com/wffh-hri/flink-pipeline-core/blob/support-2.x/src/main/scala/com/ibm/watson/health/foundation/hri/api/MgmtClient.scala

**Review by aramcodz — CHANGES_REQUESTED** [review]
NOTE: this updated version of the PR contains some implememtnation (.scala) file code that I think belongs in the 3.x release and was not a part of the support-2.x branch in our internal (github.ibm.com) version of this Repo.

File: .github/workflows/ci-workflow.yml [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        run: ./gradlew clean build
+
+#      TODO: Add SonarCloud (Static Code) scan after permissions issue is fixed
+#      - name: SonarCloud Scan
+#          uses: sonarsource/sonarcloud-github-action@v1.6
+#          with:
+#            projectBaseDir: ./src
+#          env:
+#            GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+#            SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
```
Comment (dan-baxter) [inline .github/workflows/ci-workflow.yml:52]: Left the same comment on Aram's PR -- I think this may be in a separate workflow, in which case I might remove this comment.

File: .github/workflows/ci-workflow.yml [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: ./gradlew publish
+
+      - name: Post Slack Update
+        if: ${{ job.status == 'failure' && ( github.ref == 'refs/heads/main' || github.ref == 'refs/heads/develop' || startsWith(github.ref, 'refs/heads/support-') ) }}
+        id: slack
+        uses: slackapi/slack-github-action@v1.14.0
+        with:
+          payload: "{\"Repo\":\"${{ github.repository }}\",\"Workflow\":\"${{ github.workflow }}\",\"Branch\":\"${{ env.BRANCH_NAME }}\",\"Link\":\"https://github.com/Alvearie/hri-api-spec/actions/runs/${{ github.run_id }}\"}"
```
Comment (dan-baxter) [inline .github/workflows/ci-workflow.yml:64]: Link has api-spec in it instead of hri-flink-pipeline-core

File: src/main/scala/org/alvearie/hri/api/MgmtClient.scala (hunk 1 above)
Comment (alisakotliarova) [reply src/main/scala/org/alvearie/hri/api/MgmtClient.scala]: I have reverted the `3.x` changes

File: src/main/scala/org/alvearie/hri/api/MgmtClient.scala (hunk 2 above)
Comment (alisakotliarova) [reply src/main/scala/org/alvearie/hri/api/MgmtClient.scala]: I've removed the method and compared with `2.x`, it should be matching now

File: .github/workflows/ci-workflow.yml (hunk 3 above)
Comment (alisakotliarova) [reply .github/workflows/ci-workflow.yml]: I've removed the comment ✅

File: .github/workflows/ci-workflow.yml (hunk 4 above)
Comment (alisakotliarova) [reply .github/workflows/ci-workflow.yml]: Fixed ✅

**Review by dan-baxter — APPROVED** [review]

File: .github/pull_request_template.md [hunk 5]
```diff
@@ -0,0 +1,5 @@
+## PR Requirements:
+ - [ ] The code must be reviewed by at least one Maintainer
+ - [ ] JIRA Ticket: https://jira.wh-sdlc.watson-health.ibm.com/browse/JIRA_TICKET_HERE
+ - [ ] 90 % test code coverage
+ - [ ] README updates
```
Comment (David-N-Perkins) [inline .github/pull_request_template.md:5]: I don't think you need to add this file to this branch. Only the `main` branch needs them.

File: src/main/scala/org/alvearie/hri/flink/core/MgmtApiSink.scala [hunk 6]
```diff
@@ -31,7 +31,14 @@ class MgmtApiSink(val tenantId: String, val mgmtApiUrl: String, val mgmtClientId
...
   // This enables overriding for testing
-  def createMgmtClient(): MgmtClient = new MgmtClient(mgmtApiUrl, mgmtClientId, mgmtClientSecret, mgmtClientAudience, oauthServiceBaseUrl)
+  def createMgmtClient(): MgmtClient = {
+    try {
+      new MgmtClient(mgmtApiUrl, mgmtClientId, mgmtClientSecret, mgmtClientAudience, oauthServiceBaseUrl)
+    } catch {
+      case ex: Throwable =>
+        throw new FlinkException(ex) // can't recover from this
+    }
+  }
```
Comment (David-N-Perkins) [inline src/main/scala/org/alvearie/hri/flink/core/MgmtApiSink.scala:41]: Why this change? I think this was part of changes to v3.0 that don't need to be in v2.x

**Review by aramcodz — APPROVED** [review]
Looks good to me now.

File: src/main/scala/org/alvearie/hri/flink/core/MgmtApiSink.scala (hunk 6 above)
Comment (alisakotliarova) [reply src/main/scala/org/alvearie/hri/flink/core/MgmtApiSink.scala]: Slipped by me, it's reverted now ✅

File: .github/pull_request_template.md (hunk 5 above)
Comment (alisakotliarova) [reply .github/pull_request_template.md]: Deleted ✅
